### PR TITLE
fix(git): support single character branch names

### LINKED ISSUE #3463: Single-letter branch name causes "Lexical error"
**Describe the bug**
Graph fails to render if the branch name is a single letter

**To Reproduce**
Use a single letter branch name:
```
 branch A
```

**Expected behavior**
A graph should be rendered instead of an error shown.

**Screenshots**
Broken sample (throws error):
```mermaid
gitGraph
 commit
 branch A
 commit
 checkout A

```

Broken sample (branch name given as mainBranchName parameter):
```mermaid
%%{init: { 'gitGraph': {'mainBranchName': 'A'}} }%%
gitGraph
 commit id: " " tag: "v1.0"
 branch branch1 order: 3
 checkout A
 commit id: "1st-commit"

```

Working example (graph is rendered):
```mermaid
gitGraph
 commit
 branch Ax
 commit
 checkout Ax

```

**Code Sample**
[Broken Sample](https://mermaid.live/edit#pako:eNpNjDEKwzAQBL8itvYL1AUCeUBaNefT2RL2SUY5FcH471a6dMPssCe4RoHHmu3V6EihOK6q2QbMjQon9_h3nIS32u1nMUGlKeU4Ds4xugBLohLgB0ZZqO8WEMo1UupW39_C8Na6TOhHJJNnprWRwi-0f-S6AShGMiY)

[Broken Sample with mainBranchName](https://mermaid.live/edit#pako:eNpdjs1qAzEMhF9FCBZf0tIlN99SAr3lkqsvqq3smsT24siFsvjdY-fnEhBIM3waZkWbHKPGYVh99KJhBTV5-cm0zKopFcjH70zRzgcK3Cy1U7VCHQYTX6CJYFMIXsA7DQb7CE39_Bs_vww24Pee8VwjpOw4a9j215ntORWB3XvOeJWPh9EjcIOBc-vjWt-1sY2QmQMb7LDjE5XLnawNpSLp-B8tasmFN1gWR8J7T1OmgPpElyvXG4miVJU)

[Working Sample](https://mermaid.live/edit#pako:eNpNjDEKwzAQBL8itvYL1AUCeYBbNZfT2RLxSUY5QYLx3yNSuRtmhz3ANQo81myPRnsKxXFVzTbg2ahwcrfPVXISftVuf40JKk0px3FxjNUFWBKVAD8wykJ9s4BQzpFStzp_C8Nb6zKh75FM7pnWRgq_0PaW8wel9DMW)

**Desktop (please complete the following information):**

- OS: Windows 10
- Browser Chrome Version 105.0.5195.102 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by aloisklink**
## :bookmark_tabs: Summary

In gitGraph, add support for branch names that only have a single character.

The branch regex is checking for a starting character, and an ending character, so it currently needs at least two characters.

Resolves #3463

## :straight_ruler: Design Decisions

I've wrapped everything except the first character in a `()?` to fix this.

There are some really complicated regexes that do match what valid git branches are (see https://stackoverflow.com/a/12093994), but I'm reluctant to add them in, since it will be a pain to test all the different edge-cases.

Hopefully https://github.com/mermaid-js/mermaid/pull/3432 might be used in the future to make a better gitgraph parser!

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [x] :computer: have added unit/e2e tests (if appropriate)
- [x] :bookmark: targeted `develop` branch

### CODE REVIEW

File: src/diagrams/git/parser/gitGraph.jison [hunk 1]
```diff
@@ -61,7 +61,7 @@ checkout(?=\s|$)                        return 'CHECKOUT';
 <string>["]                             this.popState();
 <string>[^"]*                           return 'STR';
 [0-9]+(?=\s|$)                          return 'NUM';
-\w[-\./\w]*[-\w]                        return 'ID'; // only a subset of https://git-scm.com/docs/git-check-ref-format
+\w([-\./\w]*[-\w])?                     return 'ID'; // only a subset of https://git-scm.com/docs/git-check-ref-format
```
Comment (knsv) [inline src/diagrams/git/parser/gitGraph.jison:64]: This makes sense

**Review by knsv — APPROVED** [review]
Thanks @aloisklink, from 
image
to
image

This makes sense!
